src/lib/serviceRoutinesV2/getRegistrations.cpp

```diff
@@ -87,7 +87,10 @@ std::string getRegistrations
     ciP->httpHeaderValue.push_back(double2string(count));
   }
 
-  TIMED_RENDER(out = vectorToJson(registrationV));
+  if (limit != 0)
+  {
+    TIMED_RENDER(out = vectorToJson(registrationV));
+  }
```

Comment:
You shouldn't do the limit=0 check here here. This is probably the cause of the problem in .test described in [previous comment](https://github.com/telefonicaid/fiware-orion/pull/3994#discussion_r783054794).

Instead, you should deal with the limit=0 condition internally at mongoRegistrationsGet() (or even in collectionRangedQuery()).

Reply:
Added condition at mongoRegistrationsGet()  in https://github.com/telefonicaid/fiware-orion/pull/3994/commits/67ea727c02d09a50bde56fcad40323bd372738c4